app-api/src/main/java/org/openmrs/eip/app/management/repository/SenderSyncMessageHistoryRepository.java

```diff
@@ -0,0 +1,58 @@
+package org.openmrs.eip.app.management.repository;
+
+import java.util.List;
+
+import org.openmrs.eip.app.management.entity.SenderSyncMessageHistory;
+import org.openmrs.eip.app.utils.CountDTO;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface SenderSyncMessageHistoryRepository extends JpaRepository<SenderSyncMessageHistory, Long> {
+	
+	@Query(value = " select table_.table_name as tableName, received as receivedItens, sent as sentItens, newer as  nonReceivedItens "
```

Comment:
Please avoid raw SQL queries, please use JPA

Reply:
I tried and it came out something very complex and with performance problems because of subqueries.